modules/effects/arrow-effects-instances/src/main/kotlin/arrow/effects/IOStart.kt

```diff
@@ -0,0 +1,49 @@
+package arrow.effects
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.core.left
+import arrow.core.right
+import arrow.effects.instances.io.applicativeError.attempt
+import arrow.effects.instances.io.async.async
+import arrow.effects.instances.io.monad.flatMap
+import arrow.effects.typeclasses.Fiber
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.startCoroutine
+import kotlin.coroutines.suspendCoroutine
+
+fun <A> IOOf<A>.startF(ctx: CoroutineContext): Kind<ForIO, Fiber<ForIO, A>> = Promise.uncancelable<ForIO, A>(IO.async()).flatMap { promise ->
+  IO.async<Fiber<ForIO, A>> { ioConnection, cb ->
+    val a: suspend () -> A = {
+      suspendCoroutine { ca: Continuation<A> ->
+        this.attempt().unsafeRunSync().fold({ error ->
```

Comment:
As it is now, this blocks the thread this coroutine runs on. Use unsafeRunAsync instead and make the callback call the paths of the continuation.

Reply:
This should be fixed now.